### PR TITLE
Remove the corresponding capabilities doc in cache when a project is removed from cache

### LINKED ISSUE #56622: QGIS Server: WMS does not notice changes of QGIS Project in PostGIS, but WFS does
### What is the bug or the crash?

When running QGIS Server with `QGIS_SERVER_PROJECT_CACHE_STRATEGY=periodic` (see [docs](QGIS_SERVER_PROJECT_CACHE_STRATEGY)), QGIS Server should react on changes of the QGIS project. This works well for WFS but **not for WMS**. 

### Steps to reproduce the issue

- start QGIS Server 3.36 with this settings (Docker compose file as example)

```
version: '3.8'
services:
  qgis_server:
    image: camptocamp/qgis-server:3.36
    volumes:
      - ./pg_service.conf:/home/.pg_service.conf
    environment:
      - QGIS_SERVER_PROJECT_CACHE_STRATEGY=periodic
      - PGSERVICEFILE=/home/.pg_service.conf
    ports:
      - "8082:80"
```

- create a QGIS project with a vector layer named `alpha` and ensure it is published both as WMS and WFS
- store this project in PostGIS
- Make a WMS and a WFS capabilities  request to the project in the database and remember the response
  - both responses should list the layer `alpha` in their response
- change the the name of the layer in your QGIS project from `alpha` to `beta`  and save the project
- Request the WMS and the WFS capabilities again. Result: 
  - WFS contains the layer called `beta` in its response
  - WMS still contains `alpha` in its response 

### Versions

3.36

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by pblottiere**
Fixes https://github.com/qgis/QGIS/issues/56622

### CODE REVIEW

File: src/server/qgscapabilitiescache.h [hunk 1]
```diff
@@ -59,6 +59,10 @@ class SERVER_EXPORT QgsCapabilitiesCache : public QObject
      */
     void removeCapabilitiesDocument( const QString &path );
 
+  public slots:
+    //! Removes changed entry from this cache
```
Comment (troopa81) [inline src/server/qgscapabilitiescache.h:63]: documentation

Comment (troopa81) [reply src/server/qgscapabilitiescache.h:63]: and maybe call directly removeCapabilitiesDocument, and get rid of dbg message

**Review by rldhont — APPROVED** [review]

**Comment by rldhont** [comment]
@pblottiere backport ?

File: src/server/qgscapabilitiescache.h (hunk 1 above)
Comment (pblottiere) [reply src/server/qgscapabilitiescache.h]: done :+1:

**Review by troopa81 — APPROVED** [review]

**Pull request merged**